### PR TITLE
Add checks before preparing sigdata and enable submit

### LINKED ISSUE #45: [xchain] handle getProof fails properly, hide SigData and disable finish button
When initial xchain transaction is completed, for several blocks, `getProof()` will fail with error message `Initial transfer is not confirmed yet. Please wait and try again.`.  

However, SigData is still prepared with `null` proof, and finish button enabled. This will lead to a failed transaction due to invalid proof. 

To prevent the failure: 
1. Hide SigData if `getProof()` fails
2. finish button disabled when `getProof()` fails

### PR DESCRIPTION
**Pull request by ggobugi27**
Closes #45

### CODE REVIEW

**Review by jmininger — APPROVED** [review]

File: docs/xchain.js [hunk 1]
```diff
@@ -88,35 +88,11 @@ const sendNonJson = async function (cmd, apiHost) {
...
-  ].reduce(async (arr, chainId) => {
-    arr = await arr;
+
+  const pactId = State.pactId;
+  const server = State.server;
+  const networkId = State.networkId;
+  const chainInfoPromises = Array.from(new Array(20)).map((_, chainId) => {
     const host = `https://${server}/chainweb/0.0/${networkId}/chain/${chainId}/pact`;
     const pactInfo = await Pact.fetch.poll({ requestKeys: [pactId] }, host);
     if (pactInfo[pactId]) {
```
Comment (alber70g) [inline docs/xchain.js:98]: This function needs to be `async`

File: docs/xchain.js [hunk 2]
```diff
@@ -362,42 +620,55 @@ function validateServer() {
...
-            if (complete()) getPact();
-            else
+            if (complete()) {
+              State.server = val;
+              fetchInfo();
+            } else {
               document
                 .getElementById('pact-message')
                 .setAttribute('class', 'ui compact message hidden');
           });
```
Comment (alber70g) [inline docs/xchain.js:635]: There's a closing curly brace missing somewhere here

File: docs/xchain.js [hunk 3]
```diff
@@ -362,42 +620,55 @@ function validateServer() {
...
+        if (complete()) {
+          await getPact();
+          await getProof();
+          await fetchInfo();
+          await completeSigData();
+        } else {
           document
             .getElementById('pact-message')
             .setAttribute('class', 'ui compact message hidden');
       } catch (err) {
```
Comment (alber70g) [inline docs/xchain.js:670]: Else-statement isn't closed

**Review by alber70g — CHANGES_REQUESTED** [review]
I suggest to run the formatter before committing. This will get syntax errors (missing braces etc) and gets it into a uniform format. 

These commands can be used if the IDE doesn't have prettier integration.

`prettier ./docs/xchain.js --check`
`prettier ./docs/xchain.js --write` (auto writes fixes to file)

**Review by alber70g — APPROVED** [review]

**Pull request merged**